### PR TITLE
Add methods for extracting Utopia and Nadir points

### PR DESCRIPTION
**Pull request by anriseth**
I started doing this ages ago but never completed it. Ref #33 
 Instead of storing the Utopia and Nadir points we can easily compute them from `Phi`. 

There were also lots of old attributes in the `MultiData` type that I have removed.

### CODE REVIEW

**Comment by anriseth** [comment]
@matbesancon would you like to look at this and merge + tag new release if you think it is a useful update?

**Comment by matbesancon** [comment]
yes sure

File: src/methods.jl [hunk 1]
```diff
@@ -30,3 +30,33 @@ function solvehook(m::Model; method::MultiMethod = WeightedSum(),
...
+
+"""
+Extract the Utopia point from the MultiData struct.
+"""
+function getutopia(m::MultiData)
+    Fmin = diag(m.Phi)
+
+    # Convert values depending on whether
+    # we maximize or minimize them.
+    sensemap = Dict(:Min => 1.0, :Max => -1.0)
```
Comment (matbesancon) [inline src/methods.jl]: building a dictionary for this seems like a lot of overhead. The suggestion below also avoids building one intermediate array
```suggestion
multiplier = (obj.sense == :Max ? -1. : 1.] for obj in m.objectives)
return [Fmin[i] * mlt for (i,mlt) in enumerate(multiplier)]
```

File: src/methods.jl [hunk 2]
```diff
@@ -30,3 +30,33 @@ function solvehook(m::Model; method::MultiMethod = WeightedSum(),
...
+"""
+function getnadir(m::MultiData)
+    Fmax = maximum(m.Phi, dims=2)
+    # Flatten array  as `maximum` returns it as a column vector.
+    # TODO: Is there a better way to do this?
+    Fmax = Fmax[:]
+
+    # Convert values depending on whether
+    # we maximize or minimize them.
+    sensemap = Dict(:Min => 1.0, :Max => -1.0)
```
Comment (matbesancon) [inline src/methods.jl]: same here

File: src/methods.jl [hunk 3]
```diff
@@ -30,3 +30,33 @@ function solvehook(m::Model; method::MultiMethod = WeightedSum(),
...
+    return Fmin .* multiplier
+end
+
+"""
+Extract the Nadir point from the MultiData struct.
+"""
+function getnadir(m::MultiData)
+    Fmax = maximum(m.Phi, dims=2)
+    # Flatten array  as `maximum` returns it as a column vector.
+    # TODO: Is there a better way to do this?
```
Comment (matbesancon) [inline src/methods.jl]: I think the following would work:
```julia
Fmax = (f for f in maximum(m.Phi, dims=2))
# multiplier as constructed above
# final for comprehension as constructed above
```

**Comment by matbesancon** [comment]
Suggested some minor improvements, even though they do not change the meaning of what is written

**Comment by anriseth** [comment]
Thanks :)

**Pull request merged**